### ISSUE TITLE
Null Pointer Exception while running Tier1 CLI suite

### ISSUE THREAD

**Issue by rarebreed** [RedHatQE/polarize#22]
I ran a tier1 CLI suite overnight and it hit an NPE.  It appears to be in the setTestSuiteResults method.  I believe the problem is that the FullResult object is null.

Here is the relevant code:

```java
       XmlTest xt = ctx.getCurrentXmlTest();
       List<XmlClass> clses = xt.getClasses();
       Set<String> clsSet = clses.stream()
               .map(x -> {
                    String sptCls = x.getSupportClass().toString();
                    sptCls = sptCls.replace("class ", "");
                    return sptCls;
               })
               .collect(Collectors.toSet());

        //..
        // While I suppose it's possible, we should have only one or zero possible results 
        // from the map, so findFirst should return at most 1.  When will we have zero?
        List<Tuple<FullResult, List<Testcase>>> frList = full.entrySet().stream()
                .filter(e -> {
                    String cls = e.getKey();
                    return clsSet.contains(cls);
                })
                .map(Map.Entry::getValue)
                .collect(Collectors.toList());
        Optional<Tuple<FullResult, List<Testcase>>> maybeFR = frList.stream().findFirst();
        Tuple<FullResult, List<Testcase>> tup = maybeFR.orElse(new Tuple<>());
        FullResult fr = tup.first;
        List<Testcase> tcs = tup.second;
        if (tcs != null)
            tests.addAll(tcs);

        setTestSuiteResults(ts, fr, ctx);
```

I believe we can get an NPE in the following scenarios:

The clsSet is empty.  If it is empty, then frList is empty.  When frList is empty, the Optional is empty, so we use the default Tuple constructor.  This should only happen if clses is empty somehow.

The filter for the frList returns no matches.  If there are no matches, frList is empty with the same result as above.  So when is the filter on frList going to return no matches?

1. If full is empty there will be no matches
2. If none of the keys in full (which are class names) are in clsSet

Finally, frList could return the first element, but the FullResult is null.  The getMethodInfo method is creating a map of String -> Tuple<FullResult, List<TestCase>>.  It is also possible that FullResult could be null if it mapped a class name to a null FullResult.

**Comment by rarebreed** [comment]
So, even if we cover all our bases, how do we even hit these scenarios?

## clsSet is empty

How would this happen?  This seems to only be possible if xt.getClasses() was null or empty somehow.

## The map from full is empty

Will only happen if getMethodInfo() returns an empty map.  This could happen if there are no invoked methods  from the class.

## None of the classname keys in full is in clsSet 

The clsSet contains all the class names run in the suite.  I could see how the clsSet could be larger than the keys in the full map, but not smaller (for example, a class that was part of the suite, but every method was skipped).  Somehow, there would have to be an invoked method whose parent class was not in clsSet.

## FullResult is null

In the findFirst method, it could return a match, but somehow the Tuple of the FullResult,List of TestCase has a null FullResult.

**Comment by rarebreed** [comment]
So it turns out it is the third option:  None of the classname keys in full is in clsSet.  What I am seeing is that there are tests like HelpTests or GeneralTests that are being iterated through from the results.entrySet(), but these classes don't exist in the full map.

Somehow, none of the methods in these classes got invoked.  Since they didn't get invoked there is no actual FullResult object

**Comment by rarebreed** [comment]
Fixed in 0788bdc0ccc4ccf88c6aae5e939f1522e58223df
